From 88ddc3cc219262ea2c3d421be0333b8625b4d77c Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Sat, 14 Dec 2024 06:48:35 +0000
Subject: [PATCH 2/2] Gate `fchown` and `fchmod` calls behind `os(WASI)`

They are not available on WASI, so we gate them behind `os(WASI)`.
---
 Sources/FoundationEssentials/FileManager/FileOperations.swift | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Sources/FoundationEssentials/FileManager/FileOperations.swift b/Sources/FoundationEssentials/FileManager/FileOperations.swift
index 9567f65..49e2c37 100644
--- a/Sources/FoundationEssentials/FileManager/FileOperations.swift
+++ b/Sources/FoundationEssentials/FileManager/FileOperations.swift
@@ -980,10 +980,12 @@ enum _FileOperations {
         #endif
         var statInfo = stat()
         if fstat(srcFD, &statInfo) == 0 {
+            #if !os(WASI) // WASI doesn't have fchown for now
             // Copy owner/group
             if fchown(dstFD, statInfo.st_uid, statInfo.st_gid) != 0 {
                 try delegate.throwIfNecessary(errno, srcPath(), dstPath())
             }
+            #endif
             
             // Copy modification date
             let value = timeval(tv_sec: statInfo.st_mtim.tv_sec, tv_usec: statInfo.st_mtim.tv_nsec / 1000)
@@ -996,10 +998,12 @@ enum _FileOperations {
                 }
             }
             
+            #if !os(WASI) // WASI doesn't have fchmod for now
             // Copy permissions
             if fchmod(dstFD, mode_t(statInfo.st_mode)) != 0 {
                 try delegate.throwIfNecessary(errno, srcPath(), dstPath())
             }
+            #endif
         } else {
             try delegate.throwIfNecessary(errno, srcPath(), dstPath())
         }
-- 
2.46.0

